### DISCUSSION TITLE
Replace LabeledTextField with html input

### DISCUSSION THREAD

**Discussion by kennycharles** [blitz-js/blitz#2984] [Q&A] [answered]
If I replace LabeledTextField with html input and submit the form, the form values are empty and zod throws an error. How do I get zod and the form to see my input?

**Comment by beerose** [comment, marked as answer]
If you look at the `LabeledTextField` component, you can see that we wrap an HTML input with form library utilities. That makes it work with the `Form` parent and validation. 

You can take a look at the `react-final-form` docs to learn more about the library, how it works, and how you can use it: https://final-form.org/docs/react-final-form/getting-started. Or, if you chose a different library during a new project initialization, you can check out the corresponding docs.
